core/src/main/java/hudson/util/MaskingClassLoader.java

```diff
@@ -62,7 +72,20 @@ public MaskingClassLoader(ClassLoader parent, Collection<String> masks) {
         return super.loadClass(name, resolve);
     }
 
+    @Override
+    public synchronized URL getResource(String name) {
```

Comment:
You need to override `getResources` too. Maybe not for this particular purpose, but for other callers.